### DISCUSSION TITLE
Custom matrix

### DISCUSSION THREAD

**Discussion by v71** [jrouwe/JoltPhysics#1401] [General]
Hello, i am new to jolt and I am about to chose between PhysX and Jolt , before starting to add it to my project , i'd like to know if its possible to supply a custom matrix to a collider object.
Thanks

**Comment by jrouwe** [comment]
I'm not quite sure what you mean with 'custom matrix to a collider object', do you mean that you want to give a shape an offset relative to a body? If so then you should use `RotatedTranslatedShape`.

**Reply by v71** [reply]
Yes, probably i didn't express myself very clearly,  I have an object which follows some matrix transformation, this object has bounding boxes to enclose more complex geometry, this bouding box is also orented accordin gto its parent matrix transformation. So i have this bounding box composed of vertices . Is it possibile to create on the fly a shape made of these vertices and test for collision only. I need only the collision because i compute the collision response according to some specific parameters ( the project is an academic simulator for drones/helycopters )

**Reply by jrouwe** [reply]
Sounds like you need to pass your vertices to a `ConvexHullShape`. And if you need collision detection only, you probably don't want to create a `Body` but just test your object for collision against the world using `PhysicsSystem::GetNarrowPhaseQuery().CollideShape(...)`. If your vertices are not centered around the origin, be sure to read the following section: https://jrouwe.github.io/JoltPhysics/#center-of-mass

**Reply by v71** [reply]
Thanks, will look into it
